presentation/src/main/java/com/stop/SoundService.kt

```diff
@@ -0,0 +1,110 @@
+package com.stop
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import android.content.Intent
+import android.media.MediaPlayer
+import android.media.RingtoneManager
+import android.os.*
+import androidx.core.app.NotificationCompat
+import androidx.lifecycle.LifecycleService
+import androidx.lifecycle.lifecycleScope
+import com.stop.domain.usecase.alarm.GetAlarmUseCase
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class SoundService : LifecycleService() {
+
+    @Inject
+    lateinit var getAlarmUseCase: GetAlarmUseCase
+
+    private var mediaPlayer: MediaPlayer? = null
+    private var vibrator: Vibrator? = null
+    private var vibratorManager: VibratorManager? = null
+
+    override fun onCreate() {
+        super.onCreate()
+
+        lifecycleScope.launch {
+            getAlarmUseCase.getAlarm().collectLatest { alarmData ->
+                alarmData?.let {
+                    if (it.alarmMethod) {
+                        mediaPlayer = MediaPlayer.create(this@SoundService, RingtoneManager.getDefaultUri(RingtoneManager.TYPE_ALARM))
+                        mediaPlayer?.setWakeMode(this@SoundService, PowerManager.PARTIAL_WAKE_LOCK)
+                        mediaPlayer?.isLooping = true
+                        mediaPlayer?.start()
```

Comment:
```suggestion
                        mediaPlayer = MediaPlayer.create(this@SoundService, RingtoneManager.getDefaultUri(RingtoneManager.TYPE_ALARM)).apply {
                            setWakeMode(this@SoundService, PowerManager.PARTIAL_WAKE_LOCK)
                            isLooping = true
                            start()
                        }
```

위와 같이 수정하면 null check를 하지 않을 수 있어요.

Reply:
e6a4ab0dfeb2c504ed72b50fdc358ebdd5891773